### PR TITLE
Adding cumsum_

### PR DESCRIPTION
**Pull request by soma2000-lang**
#10231

### CODE REVIEW

File: ivy/functional/frontends/torch/tensor.py [hunk 1]
```diff
@@ -525,6 +525,11 @@ def flatten(self, start_dim, end_dim):
     def cumsum(self, dim, dtype):
         return torch_frontend.cumsum(self._ivy_array, dim, dtype=dtype)
 
+    @with_unsupported_dtypes({"1.11.0 and below": ("float16",)}, "torch")
+    def cumsum_(self, dim, dtype):
```
Comment (guillesanbri) [inline ivy/functional/frontends/torch/tensor.py:529]: `dtype` should have `None` as the default value (https://pytorch.org/docs/1.11/generated/torch.Tensor.cumsum_.html?highlight=cumsum#torch.Tensor.cumsum_)

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -3676,6 +3676,48 @@ def test_torch_instance_cumsum(
...
+    dim=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(), key="shape"),
+        allow_neg=True,
+        force_int=True,
+    ),
+)
+def test_torch_instance_cumsum_(
+    dtype_value,
+    dim,
+    dtypes,
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3697]: seems like this is already generated in `dtype_value`

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 3]
```diff
@@ -3676,6 +3676,48 @@ def test_torch_instance_cumsum(
...
+    input_dtype, x = dtype_value
+    helpers.test_frontend_method(
+        init_input_dtypes=input_dtype,
+        init_all_as_kwargs_np={
+            "data": x[0],
+        },
+        method_input_dtypes=dtypes,
+        method_all_as_kwargs_np={
+            "dim": dim,
+            "dtype": dtypes[0],
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3712]: `dtypes` is not being generated, instead, use `input_dtype`

**Review by guillesanbri — CHANGES_REQUESTED** [review]
Hi, thanks for the contribution! I've left a couple of comments requesting changes, let me know if you need anything!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py (hunk 2 above)
Comment (soma2000-lang) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3698]: @guillesanbri So we need to remove dtype_value from the argument right?

Comment (guillesanbri) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3697]: Actually, `dtype` has to be removed from the args. You are generating `dtype_value` and `dim` in the decorator (which is fine), but the function is expecting `dtype`, `dim` and `dtypes`

**Comment by soma2000-lang** [comment]
@guillesanbri  sorry for the trouble,made the changes but still failing!

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 4]
```diff
@@ -3676,6 +3676,47 @@ def test_torch_instance_cumsum(
     )
 
 
+# cumsum_
+@handle_frontend_method(
+    class_tree=CLASS_TREE,
+    init_tree="torch.tensor",
+    method_name="cumsum_",
+    dtype_value=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes("valid"),
```
Comment (guillesanbri) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3685]: seems like the ground truth does not support `bool`, you can change "valid" to "numeric" here

File: ivy/functional/frontends/torch/tensor.py [hunk 5]
```diff
@@ -525,6 +525,11 @@ def flatten(self, start_dim, end_dim):
     def cumsum(self, dim, dtype):
         return torch_frontend.cumsum(self._ivy_array, dim, dtype=dtype)
 
+    @with_unsupported_dtypes({"1.11.0 and below": ("float16",)}, "torch")
+    def cumsum_(self, dim, dtype=None):
```
Comment (guillesanbri) [inline ivy/functional/frontends/torch/tensor.py:529]: `dtype` has to be a keyword argument, in order for the test to detect this automatically, you should add `*,` between `dim` and `dtype` (i.e. `def cumsum_(self, dim, *, dtype=None):`)

**Review by guillesanbri — CHANGES_REQUESTED** [review]
just left a couple of changes more, should be good to go after this! thanks!

**Comment by soma2000-lang** [comment]
@guillesanbri Done! Thanks

**Comment by guillesanbri** [comment]
lgtm! I'll merge it now, thanks for the contribution! :smile:
